cmd/axelard/cmd/vald/tss/sign.go

```diff
@@ -226,9 +226,16 @@ func (mgr *Mgr) setSignStream(sigID string, stream Stream) {
 
 func (mgr *Mgr) multiSigSignStart(keyID string, sigID string, shares uint32, payload []byte) error {
 	var signatures [][]byte
+	pubKey := []byte{}
+	pubKeys, found := mgr.getPubKeys(tssexported.KeyID(keyID))
+
 	for i := uint32(0); i < shares; i++ {
 		keyUID := fmt.Sprintf("%s_%d", keyID, i)
-		signature, err := mgr.multiSigSign(keyUID, payload)
+		if found {
+			pubKey = pubKeys[i]
+		}
+
+		signature, err := mgr.multiSigSign(keyUID, payload, pubKey)
```

Comment:
same here. Why not just skip the whole thing is vald doesn't have anything to do with the keyID?

Reply:
Also I don't quite understand why does vald needs to keep track of the public keys it's tofnd is supposed to have. Does tofnd not store keyUID to private key mapping?

Reply:
I was concerned about some edge cases that would cause it to miss signing, but in those cases it would have caused vald to be killed and restarted in which case a fresh mapping would have been queried anyways. So, I think it makes sense to return an error if a sign event is received but a corresponding keyID wasn't found.

Yeah, we decided to not do that in tofnd to keep recovery very simple. And if we did do it, vald would still have to query this info, and help tofnd during recovery which would add more complexity.